### ISSUE TITLE
Deprecation warnings with ruby 2.7

### ISSUE THREAD

**Issue by sriedel** [Asana/ruby-asana#88]
With ruby 2.7 a more strict separation of hash and keyword arguments was introduced along with aggressive deprecation notifications.

With the ruby-asana gem version 0.9.3 and ruby 2.7.0 I'm getting:

    /.../lib/ruby/gems/2.7.0/gems/asana-0.9.3/lib/asana/client.rb:63: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
    /.../lib/ruby/gems/2.7.0/gems/asana-0.9.3/lib/asana/resources/task.rb:99: warning: The called method `create' is defined here

The culprit is a method_missing definition, and at first glance I'm not sure what the correct fix would be.

**Comment by chrismo** [comment]
The version of faraday y'all use is also generating deprecation warnings. There's a version 1.0.0 of the gem that's been released which addresses this, but the requirement is ~> 0.9, so it would need to be relaxed to allow the newer version.

**Comment by rossgrambo** [comment]
Perhaps one of you could direct me how to get these warnings. If i create a new project, gem install asana and run:

```
require('asana')

client = Asana::Client.new do |c|
  c.authentication :access_token, ENV["PAT"]
end

me = client.users.me
```

I'm not given any deprecation warnings. With or without opting into the flag `-W:deprecated`.

I know what changes I need to make, but I'd prefer to see the deprecation warnings myself before making the changes.

**Comment by sriedel** [comment]
For me the warnings appear when I call

    client.tasks.create(workspace: 'test', **data)

**Comment by chrismo** [comment]
Looks like this is fixed now: https://github.com/Asana/ruby-asana/commit/f67c3dc319fe34a8c401afed10c037bf81a1939b

**Comment by sriedel** [comment]
No it's not. 

While the faraday version got updated, that wasn't the original intent of this bugreport.
The deprecation warnings are still there with asana 0.10.1:
    /.../lib/ruby/gems/2.7.0/gems/asana-0.10.1/lib/asana/client.rb:63: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
    /.../lib/ruby/gems/2.7.0/gems/asana-0.10.1/lib/asana/resources/task.rb:96: warning: The called method `create' is defined here

**Comment by rossgrambo** [comment]
This should finally be resolved in v0.10.2. Let me know if this issue persists.

**Comment by sriedel** [comment]
Just updated the relevant code base on friday, and no more warnings :)

Thanks!
